src/server/common/helpers/auth/azure-oidc.js

```diff
@@ -79,13 +79,15 @@ const azureOidc = {
         refreshAccessToken(token, wellKnown)
       )
 
-      server.ext('onPreAuth', (request, h) =>
-        refreshTokenIfExpired(
+      server.ext('onPreAuth', async (request, h) => {
+        const userSession = await request.getUserSession()
```

Comment:
worth adding a comment that this is in `onPreAuth` hence we are having to get the session from the cache via the `getUserSession` helper?